types.go

```diff
@@ -521,3 +519,117 @@ func ParseAutocompletePlaceType(placeType string) (AutocompletePlaceType, error)
+	PlaceDetailsFieldMaskWebsite                  = PlaceDetailsFieldMask("website")
+	PlaceDetailsFieldMaskFormattedPhoneNumber     = PlaceDetailsFieldMask("formatted_phone_number")
+	PlaceDetailsFieldMaskInternationalPhoneNumber = PlaceDetailsFieldMask("international_phone_number")
+	PlaceDetailsFieldMaskPriceLevel               = PlaceDetailsFieldMask("price_level")
+	PlaceDetailsFieldMaskRatings                  = PlaceDetailsFieldMask("rating")
+	PlaceDetailsFieldMaskReviews                  = PlaceDetailsFieldMask("reviews")
+)
+
+// ParsePlaceDetailsFieldMask will parse a string representation of
+// PlaceDetailsFieldMask.
+func ParsePlaceDetailsFieldMask(placeDetailsFieldMask string) (PlaceDetailsFieldMask, error) {
+	switch strings.ToLower(placeDetailsFieldMask) {
+	case "place_id":
+		return PlaceDetailsFieldMaskPlaceID, nil
+	case "name":
+		return PlaceDetailsFieldMaskName, nil
+	case "type":
+		return PlaceDetailsFieldMaskType, nil
+	case "address_components":
+		return PlaceDetailsFieldMaskAddressComponents, nil
+	case "formatted_address":
+		return PlaceDetailsFieldMaskFormattedAddress, nil
+	case "url":
+		return PlaceDetailsFieldMaskURL, nil
+	case "utc_offset":
+		return PlaceDetailsFieldMaskUTCOffset, nil
+	case "permanently_closed":
+		return PlaceDetailsFieldMaskPermanentlyClosed, nil
+	case "geometry":
+		return PlaceDetailsFieldMaskGeometry, nil
+	case "geometry.location":
+		return PlaceDetailsFieldMaskGeometryLocation, nil
+	case "geometry.viewport":
+		return PlaceDetailsFieldMaskGeometryViewport, nil
+	case "photo":
+		return PlaceDetailsFieldMaskPhoto, nil
+	case "photo.photo_reference":
+		return PlaceDetailsFieldMaskPhotoReference, nil
+	case "photos":
+		return PlaceDetailsFieldMaskPhotos, nil
+	case "icon":
+		return PlaceDetailsFieldMaskIcon, nil
+	case "types":
+		return PlaceDetailsFieldMaskTypes, nil
+	case "adr_address":
+		return PlaceDetailsFieldMaskADRAddress, nil
+	case "scope":
+		return PlaceDetailsFieldMaskScope, nil
+	case "vicinity":
+		return PlaceDetailsFieldMaskVicinity, nil
+	case "opening_hours":
+		return PlaceDetailsFieldMaskOpeningHours, nil
+	case "opening_hours.weekday_text":
+		return PlaceDetailsFieldMaskOpeningHoursWeekdayText, nil
+	case "opening_hours.open_now":
+		return PlaceDetailsFieldMaskOpeningHoursOpenNow, nil
+	case "opening_hours.period":
+		return PlaceDetailsFieldMaskOpeningHoursPeriod, nil
+	case "website":
+		return PlaceDetailsFieldMaskWebsite, nil
+	case "formatted_phone_number":
+		return PlaceDetailsFieldMaskFormattedPhoneNumber, nil
+	case "international_phone_number":
+		return PlaceDetailsFieldMaskInternationalPhoneNumber, nil
+	case "price_level":
+		return PlaceDetailsFieldMaskPriceLevel, nil
+	case "rating":
+		return PlaceDetailsFieldMaskRatings, nil
+	case "reviews":
+		return PlaceDetailsFieldMaskReviews, nil
+	default:
+		return PlaceDetailsFieldMask("Unknown PlaceDetailsFieldMask"), fmt.Errorf("Unknown PlaceDetailsFieldMask \"%v\"", placeDetailsFieldMask)
```

Comment:
should this return "" as the type? It seems weird that it might be possible to use "Unknown PlaceDetailsFieldMask" as an actual mask. Zero values make more sense when returning an error.